### PR TITLE
Add tests for provider/common

### PR DESCRIPTION
**Pull request by aojea**
Add testing for the sigs.k8s.io/kind/pkg/internal/cluster/providers/provider/common package

```
Running tool: /usr/local/go/bin/go test -timeout 30s sigs.k8s.io/kind/pkg/internal/cluster/providers/provider/common -coverprofile=/tmp/vscode-gozIao1h/go-code-cover

ok  	sigs.k8s.io/kind/pkg/internal/cluster/providers/provider/common	0.003s	coverage: 97.1% of statements
Success: Tests passed.
```

### CODE REVIEW

File: pkg/internal/cluster/providers/provider/common/proxy_test.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+				return &c
+			}(),
+			envs: map[string]string{"HTTP_PROXY": "5.5.5.5"},
+			want: map[string]string{"HTTP_PROXY": "5.5.5.5", "http_proxy": "5.5.5.5", "NO_PROXY": ",10.0.0.0/24,12.0.0.0/24", "no_proxy": ",10.0.0.0/24,12.0.0.0/24"},
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			os.Clearenv()
```
Comment (BenTheElder) [inline pkg/internal/cluster/providers/provider/common/proxy_test.go]: we should avoid mutating global state, I can write a follow-up to fix this though.

Comment (aojea) [reply pkg/internal/cluster/providers/provider/common/proxy_test.go]: I'm happy to fix it if you just guide me

Comment (BenTheElder) [reply pkg/internal/cluster/providers/provider/common/proxy_test.go]: we should make `GetProxyEnvs()` a thin wrapper around a method with an injected dependency.

Comment (aojea) [reply pkg/internal/cluster/providers/provider/common/proxy_test.go]: PTAL, I've used this approach
https://github.com/kubernetes/kubernetes/blob/master/cluster/images/conformance/go-runner/env.go

File: pkg/internal/cluster/providers/docker/provision.go [hunk 2]
```diff
@@ -209,7 +209,8 @@ func runArgsForLoadBalancer(cfg *config.Cluster, name string, args []string) ([]
 }
 
 func getProxyEnv(cfg *config.Cluster) (map[string]string, error) {
-	envs := common.GetProxyEnvs(cfg)
+	getEnv := &common.OsEnv{}
+	envs := common.GetProxyEnvs(cfg, getEnv)
```
Comment (BenTheElder) [inline pkg/internal/cluster/providers/docker/provision.go]: the caller shouldn't be concerned with this.

File: pkg/internal/cluster/providers/provider/common/proxy.go [hunk 3]
```diff
@@ -34,12 +33,12 @@ const (
 
 // GetProxyEnvs returns a map of proxy environment variables to their values
 // If proxy settings are set, NO_PROXY is modified to include the cluster subnets
-func GetProxyEnvs(cfg *config.Cluster) map[string]string {
+func GetProxyEnvs(cfg *config.Cluster, env Getenver) map[string]string {
```
Comment (BenTheElder) [inline pkg/internal/cluster/providers/provider/common/proxy.go]: the public API doesn't need changing, just make the public a wrapper around a private method that passes in the real getenv

File: pkg/internal/cluster/providers/provider/common/env.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+package common
+
+import (
+	"os"
+)
+
+// Getenver is the interface we use to mock out the env for easier testing. OS env
+// vars can't be as easily tested since internally it uses sync.Once.
+type Getenver interface {
+	Getenv(string) string
```
Comment (BenTheElder) [inline pkg/internal/cluster/providers/provider/common/env.go]: this is overkill, we can just pass in a `func(string)(string)`

File: pkg/internal/cluster/providers/provider/common/proxy.go (hunk 3 above)
Comment (BenTheElder) [reply pkg/internal/cluster/providers/provider/common/proxy.go]: I implemented this in https://github.com/kubernetes-sigs/kind/pull/969

**Comment by BenTheElder** [comment]
rebased commits merged in https://github.com/kubernetes-sigs/kind/pull/969
thanks again!

**Pull request closed without merging**